contracts/test/Proofs.t.sol

```diff
@@ -68,7 +78,7 @@ contract MerkleProofTest is Test {
     // Builds a merkle tree from an array of leaves.
     // The tree is an array of arrays of bytes32.
     // The last array is the leaves, and each prior array is the result of the commutative hash of pairs in the previous array.
-    // An unpaired element is simply copied to the next level.
+    // An unpaired element is paired with itself to create the value at the next level up.
```

Comment:
My primary concern was simplicity and security. 

The old behaviour of promoting the unpaired element to the next layer (is that what you mean by identity hash?), thus shortening the proof, broke tracking of the position while verifying proofs. I did a little reading about how to handle odd nodes in this situation, and pairing with self (thus generating a new hash at each level, rather than it being predictable from just the leaf, or pairing with a constant) seemed recommended.

An alternatives we could look at:
- While verifying, given known width of the tree, track whether or not the last element is unpaired at each level and skip checking if our position is that last element. This would add some complexity to avoid one extra hash at each level (log in data size) while building the tree.

We should certainly seek some cryptographer review as we settle on things here.